.github/workflows/e2e.yml

```diff
@@ -0,0 +1,123 @@
+        id: filter
+        with:
+          working-directory: plugin
+          filters: |
+            core-plugins:
+              - 'core-plugins/**'
+      - name: Set modules list
+        id: sml
+        run: |
+          if [ ${{ github.event_name }} != 'workflow_dispatch' ]; then
+            echo "::set-output name=matrix::$( jq '.' --compact-output <<< '${{ steps.filter.outputs.changes }}')"
+          else
+            echo "::set-output name=matrix::['core-plugins']"
+          fi
+        shell: bash
+    outputs:
+      matrix: ${{ steps.sml.outputs.matrix }}
+
+  build:
+    needs: modules
+    runs-on: k8s-runner-e2e
+    if: needs.modules.outputs.matrix != '[]' && needs.modules.outputs.matrix != ''
+    strategy:
+      matrix:
+        module: ${{ fromJson(needs.modules.outputs.matrix) }}
+    steps:
+      # Pinned 1.0.0 version
+      - uses: actions/checkout@v2.3.4
+        with:
+          path: plugin
+          submodules: 'recursive'
+      - name: Checkout e2e test repo
+        uses: actions/checkout@v2.3.4
+        with:
+          repository: cdapio/cdap-e2e-tests
+          path: e2e
+      - name: Cache
+        uses: actions/cache@v2.1.3
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ github.workflow }}-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-${{ github.workflow }}
+      - name: Run required e2e tests
+        if: github.event_name != 'workflow_dispatch' && (!contains( github.event.head_commit.message, 'Merge pull request'))
+        run: python3 e2e/src/main/scripts/run_e2e_test.py --module ${{ matrix.module }} --testRunner TestRunnerRequired.java
+      - name: Run all e2e tests
+        if: github.event_name == 'workflow_dispatch' || (contains( github.event.head_commit.message, 'Merge pull request'))
+        run: python3 e2e/src/main/scripts/run_e2e_test.py --module ${{ matrix.module }}
+      - name: Upload report
+        uses: actions/upload-artifact@v2.2.4
+        if: always()
+        with:
+          name: Cucumber report
+          path: ./**/target/cucumber-reports
+      - name: Upload debug files
+        uses: actions/upload-artifact@v2.2.4
+        if: always()
+        with:
+          name: Debug files
+          path: ./**/target/e2e-debug
+      - name: Upload files to GCS
+        uses: google-github-actions/upload-cloud-storage@v0
+        if: always()
+        with:
+          path: ./plugin
+          destination: e2e-tests-cucumber-reports/${{ github.event.repository.name }}/${{ github.ref }}
+          glob: '**/target/cucumber-reports/**'
+      - name: github-status-action
+        uses: Sibz/github-status-action@67af1f4042a5a790681aad83c44008ca6cfab83d
+        if: always()
+        with:
+          authToken: ${{ secrets.GITHUB_TOKEN }}
+          state: success
+          context: Cucumber report
+          sha: ${{github.event.pull_request.head.sha || github.sha}}
```

Comment:
As we are not publishing cucumber reports back, can remove this step for now.

Reply:
Removed step.